packages/amplify_api_plugin_interface/lib/src/types/pagination/paginated_result.dart

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// TODO: Datastore dependencies temporarily added in API. Eventually they should be moved to core or otherwise reconciled to avoid duplication.
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+class PaginatedResult<T extends Model> {
+  @override
+  String getId() {
```

Comment:
Prefer getters over `getX()` methods, for example:

```dart
String get id;
```

instead of

```dart
String getId();
```

Reply:
These methods aren't overriding anything, so we can remove `@override`

Reply:
Actually, this can just be an abstract class since all it is doing is setting up an interface for subclasses to override.

```dart
abstract class PaginatedResult<T extends Model> {
  String get id;

  ModelType<T> get instanceType;

  Map<String, dynamic> toJson();
}
```

Reply:
Great feedback, totally missed this not being an abstract. Fixed 👍